blog/2024-11-linked-signal/README.md

```diff
@@ -0,0 +1,266 @@
+This pattern is useful in forms where we want fields to reset to a default state when certain inputs change.
+
+## Advanced Scenarios for Linked Signals
+
+### Nested State Management
+
+Suppose you have nested data such as book properties (`title` and `rating`), and we want these fields to reset when a different `book` is selected. 
+Here’s how we could manage this with a Linked Signal:
+
+```typescript
+import { Component, input, linkedSignal } from '@angular/core';
+
+@Component({
+  selector: 'app-book',
+  template: `
+    <p>Title: {{ title() }}</p>
+    <p>Rating: {{ rating() }}</p>
+
+    <button (click)="doRateUp()">Rate up</button>
+  `,
+})
+class BookComponent  {
+  book = input.required<Book>();
+  ratingChange = output<{ isbn: string, newRating: number }>();
+
+  title = computed(() => this.book().title);
+
+  rating = linkedSignal({
+    source: this.book,
+    computation: book => book.rating,
+  });
+
+  doRateUp() {
+    const newRating = this.rating() + 1;
+    this.rating.set(newRating);
+
+    this.ratingChange.emit({ 
+      isbn: this.book().isbn,
+      newRating
+    });
+  }
+}
+```
+
+With this setup, both `title` and `rating` reset when `book` changes, helping to keep data synchronized in cases where the structure of state is hierarchical or dependent on specific identifiers. 
+While the Linked Signal makes sure that the data resets when necessary, we can still update our local state directly. 
+In this example we update `rating` locally and communicate the change back to the parent component.
+Since we don’t need to modify the `title` within the component, a simple computed signal fulfils this task.
+
+
+### Synchronizing Server-Data for Client-Side Edits
+
+A Linked Signal is also helpful when working with server data that needs to be edited locally.
+If we’re fetching data from an API but need to allow changes on the client side, we can use `linkedSignal()` to keep local edits in sync with the original server data.
+Here is an example that uses data from our HTTP API, fetched through a simple `httpClient` wrapper called `BookStoreService`:
+
+```typescript
+import { Component, inject, linkedSignal } from '@angular/core';
+import { BookStoreService } from './book-store.service';
+
+@Component({
+  selector: 'app-dashboard',
+  template: `
+    @for (b of books(); track b.isbn) {
+      <app-book
+        (ratingChange)="handleRatingChange($event)"
+        [book]="b"
+      />
+    } 
+
+    <button (click)="changeOrder()">Change order (locally)</button>
+  `,
+})
+export class DashboardComponent {
+  private bookStore = inject(BookStoreService);
+
+  books = linkedSignal(
+    toSignal(this.bookStore.getAllBooks(), { initialValue: [] })
+  );
+
+  changeOrder() {
+    this.books.update(books => books.toReversed());
+  }
+
+  handleRatingChange({ isbn, newRating }: { isbn: string, newRating: number }) {
+    this.books.update(books => books.map(b => b.isbn === isbn ? {...b, rating: newRating } : b));
```

Comment:
Can we write this in a full block with real if/else and return? I understand what it does but the blog post is not about hoe to upste a book list, so I'd like to keep this as simple and readable as possible.